ulib/experimental/Steel.Channel.Simplex.fst

```diff
@@ -448,7 +448,10 @@ let channel_cases (r:ref chan_val) (#p:prot{more p}) #q (c:chan q) (x:msg_t p) (
     = let cc = c.chan_chan in
       h_assert (send_pre r p cc vs vr);
       h_assert (send_pre_split r p cc vs vr (vs.chan_ctr = vr.chan_ctr));
-      cond (vs.chan_ctr = vr.chan_ctr) (send_pre_split r p cc vs vr) _ then_ else_
+      let b = vs.chan_ctr = vr.chan_ctr in
+      if b returns SteelT unit (send_pre_split r p cc vs vr b) _ //(fun _ -> in_state r (step p x))
+      then then_ ()
+      else else_ ()
```

Comment:
Would be even nicer if we could simply remove the channel_cases function and just use a native if/returns in place. I can try.